### ISSUE TITLE
TypeError on Loading MarkerClusterDirective

### ISSUE THREAD

**Issue by dboschm** [Asymmetrik/ngx-leaflet-markercluster#21]
Hi. I want to use ngx-leaflet-markercluster in an Ionic 4 app with Angular 7.2.2.
I get the following error on loading the MarkerClusterDirective:
```
ERROR TypeError: L.MarkerClusterGroup is not a constructor
    at Object.push../node_modules/leaflet.markercluster/dist/leaflet.markercluster-src.js.L.markerClusterGroup (leaflet.markercluster-src.js:1385)
    at LeafletMarkerClusterDirective.push../node_modules/@asymmetrik/ngx-leaflet-markercluster/dist/leaflet-markercluster/leaflet-markercluster.directive.js.LeafletMarkerClusterDirective.ngOnInit (leaflet-markercluster.directive.js:25)
    at checkAndUpdateDirectiveInline (core.js:22098)
    at checkAndUpdateNodeInline (core.js:23362)
    at checkAndUpdateNode (core.js:23324)
    at debugCheckAndUpdateNode (core.js:23958)
    at debugCheckDirectivesFn (core.js:23918)
    at Object.eval [as updateDirectives] (LeafletMapComponent.html:1)
    at Object.debugUpdateDirectives [as updateDirectives] (core.js:23910)
    at checkAndUpdateView (core.js:23306)
```

My component looks like the demo file:
```
<div id="map" 
  leaflet 
  [leafletOptions]="options" 
  (leafletMapReady)="onMapReady($event)"
  
  [leafletMarkerClusterOptions]="markerClusterOptions"
  [leafletMarkerCluster]="markerClusterData" 
  (leafletMarkerClusterReady)="markerClusterReady($event)">
</div>
```

On init `markerClusterData` is an empty Array `[]`. 
As `markerClusterOptions` I use the default options from `leaflet.markercluster`

I have installed the latest packages as well as the latest types versions.
```
dependancies
    "@asymmetrik/ngx-leaflet": "^5.0.1",
    "@asymmetrik/ngx-leaflet-markercluster": "^2.0.0",
    "leaflet": "^1.4.0",
    "leaflet.markercluster": "^1.4.1",

devDependancies
    "@types/leaflet": "^1.4.3",
    "@types/leaflet.markercluster": "^1.4.0",
    "typescript": "~3.1.6"
```

I have absolutely no Idea how to fix this problem.

**Comment by reblace** [comment]
Can you share your component code?

**Comment by reblace** [comment]
With the same versions you list, I was able to get a basic example working with Angular CLI. If you could provide a full example as a git repo I could try to suggest a fix. Otherwise, I'm unable to reproduce this issue.

**Comment by dboschm** [comment]
Here are the relevant parts of my Component.
```
@Component({
  selector: 'app-leaflet-map',
  templateUrl: './leaflet-map.component.html',
  styleUrls: ['./leaflet-map.component.scss'],
})
export class LeafletMapComponent implements OnInit {

// leafletOptions
options = {
    layers: [
      L.tileLayer('http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
        maxZoom: 19,
        attribution: '...',
        detectRetina: true,
        updateWhenZooming: true,
        tileSize: 512,
        zoomOffset: -1,
      }),
    ],
    zoom: 5,
    center: L.latLng(46.879966, -121.826909),
  };
...

// Markers are updated via WebSockets
this.markerService.getMarkerWsObservable().subscribe((markerDto: MarkerDto) => {
      const iconId = markerDto.icon;
      const vectorIcon = new vm.VectorMarkers.Icon({
        extraClasses: iconId.style,
        icon: iconId.icon,
        prefix: 'fa',
        markerColor: '#FF5733',
        iconColor: 'white'
      });
      // random dummy positions
      new L.Marker([this.generateLat(), this.generateLon()], { icon: vectorIcon })
        .addTo(this.map);
    });

generateLat() { return Math.random() * 360 - 180; }
generateLon() { return Math.random() * 180 - 90; }
```

**Comment by dboschm** [comment]
Here are the relevant parts of my Component.
```
import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
import * as L from 'leaflet';
import * as vm from 'Leaflet.vector-markers';
import { MarkerService, MarkerDto } from 'src/app/_shared/marker.service';

@Component({
  selector: 'app-leaflet-map',
  templateUrl: './leaflet-map.component.html',
  styleUrls: ['./leaflet-map.component.scss'],
})
export class LeafletMapComponent implements OnInit {

markerClusterData = [];
markerClusterOptions: L.MarkerClusterGroupOptions;

options = {
    layers: [
      L.tileLayer('http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
        maxZoom: 19,
        attribution: '...',
        detectRetina: true,
        updateWhenZooming: true,
        tileSize: 512,
        zoomOffset: -1,
      }),
    ],
    zoom: 5,
    center: L.latLng(46.879966, -121.826909),
  };
...

// Markers are updated via WebSockets
this.markerService.getMarkerWsObservable().subscribe((markerDto: MarkerDto) => {
      const iconId = markerDto.icon;
      const vectorIcon = new vm.VectorMarkers.Icon({
        extraClasses: iconId.style,
        icon: iconId.icon,
        prefix: 'fa',
        markerColor: '#FF5733',
        iconColor: 'white'
      });
      // random dummy positions
      const marker = new L.Marker([this.generateLat(), this.generateLon()], { icon: vectorIcon })
        .addTo(this.map);
      this.markerClusterData.push(marker);
    });

generateLat() { return Math.random() * 360 - 180; }
generateLon() { return Math.random() * 180 - 90; }
```

**Comment by reblace** [comment]
I started with a new ionic app and was unable to reproduce this issue. Without seeing a complete example that reproduces the issue, there isn't much more I can do to help.

The one thing that tends to trip people up is that they haven't imported the `LeafletModule` and `LeafetMarkerClusterModule` directly into both the app module as well as the local module in which the libraries are being used. But, that wouldn't typically cause the issue you're seeing.

It's also possible that I've fixed something in the latest minor release of ngx-leaflet-markercluster. So, you might try upgrading to see if it addresses the issue.

These are the dependencies from my package.json:

```
"leaflet": "^1.5.1",
"leaflet.markercluster": "^1.4.1",

"@types/leaflet": "^1.4.4",
"@types/leaflet.markercluster": "^1.4.0",

"@asymmetrik/ngx-leaflet": "^5.0.2",
"@asymmetrik/ngx-leaflet-markercluster": "^2.1.1",
```
